packages/html-entities/src/index.js

```diff
@@ -36,5 +37,6 @@ export function decodeEntities( html ) {
 	_decodeTextArea.innerHTML = html;
 	const decoded = _decodeTextArea.textContent;
 	_decodeTextArea.innerHTML = '';
-	return decoded;
+	// Cast to string, HTMLTextElement should always have `string` textContent
```

Comment:
It wasn’t immediately clear to me why we’d need the cast. I found the relevant documentation, which perhaps this content was based on:

https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent#Description

Could we include a link, and maybe even some of the quoted description?

Odd to me that if we type the field as a textarea that it could ever consider this as being anything other than a string. I guess it may just not be implemented in the TypeScript DOM libraries to make a distinction on that point?

Reply:
I've extended the comment with more details and a link 👍 

> Odd to me that if we type the field as a textarea that it could ever consider this as being anything other than a string. I guess it may just not be implemented in the TypeScript DOM libraries to make a distinction on that point?

I don't know how much TypeScript libs try to follow spec, or how important the spec is. These statements taken together:

> The `textContent` property of the `Node` interface represents the text content of the node and its descendants.
> ## Value
> A string or `null`

Clearly describe the interface:

```ts
interface Node {
  // …
  textContent: string | null;
}
```

which is [what we find](https://github.com/microsoft/TypeScript/blob/ef377d5a66266a230ea5a06bc73fac43cfc06374/lib/lib.dom.d.ts#L11046).

However, the MDN description suggests that (roughly) the following would be accurate as well:

```ts
interface Document extends Node {
  textContent: null;
}
interface HTMLElement extends Node {
  textContent: string;
}
```

but we don't find that 🤷‍♂